namespaces/mson-namespace.md

```diff
@@ -0,0 +1,503 @@
+# MSON Namespace
+
+This document extends [Refract][]'s [JSON Namespace][] with elements necessary to
+build [MSON][] DOM.
+
+# Content
+
+This namespace defines following elements:
+
+1. General-use elements
+    1. [Id Element](#id-element-distinct-element)
+    1. [Distinct Element](#distinct-element-element)
+    1. [Select](#select-distinct-element)
+    1. [Option](#option-distinct-element)
+
+1. MSON DOM-specific elements
+    1. [MSON Element](#mson-element-distinct-element)
+    1. [Boolean Type](#boolean-type-jsonboolean-type)
+    1. [String Type](#string-type-jsonstring-type)
+    1. [Number Type](#number-type-jsonnumber-type)
+    1. [Array Type](#array-type-jsonarray-type)
+    1. [Object Type](#object-type-jsonobject-type)
+    1. [Property Type](#property-type-jsonproperty-type)
+    1. [Enum Type](#enum-type-mson-element)
+
+# Terminology
+
+This specification uses terminology from [MSON Specification][], in addition, bellow you can find the explanation of terms defined in this namespace.
+
+## Expanded Element
+
+MSON is built around the idea of defining recursive data structures. To provide abstraction, for convenience reasons and to not repeat ourselves these structures can be named (using an _identifier_) and reused. In MSON, the reusable data structures are called _Named Types_.
```

Comment:
repeat ourselves,

In \[MSON][], reusable ...

or

In \[MSON][], these reusable ...